File: typescript/common/todo.manager.ts
import { CosmosClient } from "@azure/cosmos";
import { Guid } from "guid-typescript";
import ITodoManager from './itodo.manager';

export default class TodoManager implements ITodoManager {
    private todoDbId: string = process.env.TodoDbId;
    private todoDbContainerId: string = process.env.TodoDbContainerId;

    public constructor(private readonly client: CosmosClient) {}

    public async addTodoAsync(todo: ITodo): Promise<string> {
        const container = this.client.database(this.todoDbId).container(this.todoDbContainerId);
        todo.id = Guid.create().toString();
        await container.items.create(todo);
        return todo.id;
    }

    public async getTodosAsync(): Promise<ITodo[]> {
        const container = this.client.database(this.todoDbId).container(this.todoDbContainerId);
        const { resources } = await container.items.query('SELECT * FROM c').fetchAll();
        return this.toTodos(resources);
    }

    private toTodos(resources: any[]): ITodo[] {
        const todos: ITodo[] = [];
        for (const item of resources) {
            todos.push({ id: item.id, value: item.value });
        }
        return todos;
    }
}

File: csharp/TodoManager.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.Extensions.Options;
using Azure.Cosmos;
using System;

namespace GottaBeGeek
{
    public class TodoManager : ITodoManager
    {
        private readonly TodoOptions options;
        private readonly CosmosClient cosmosClient;

        public TodoManager(IOptions<TodoOptions> options)
        {
            this.options = options?.Value;
            cosmosClient = new CosmosClient(this.options.TodoDbUrl, this.options.TodoDbKey);
        }

        public async Task<Guid> AddTodoAsync(Todo todo)
        {
            todo.Id = Guid.NewGuid().ToString();
            var container = cosmosClient.GetContainer(options.TodoDbId, options.TodoDbContainerId);
            var response = await container.CreateItemAsync<Todo>(todo, new PartitionKey(todo.Id)).ConfigureAwait(false);
            if (response.Value == null)
            {
                throw new Exception("Creation Failed");
            }
            return Guid.Parse(response.Value.Id);
        }

        public async Task<IEnumerable<Todo>> GetTodosAsync()
        {
            var container = cosmosClient.GetContainer(options.TodoDbId, options.TodoDbContainerId);

            var query = new QueryDefinition("SELECT * FROM c");

            var todos = new List<Todo>();
            await foreach (var todo in container.GetItemQueryIterator<Todo>(query))
            {
                todos.Add(todo);
            }
            return todos;
        }
    }
}